app/Repository/Eloquent/SavedLearningItemRepository.php

```diff
@@ -30,4 +33,13 @@ public function save(SavedLearningItem $savedLearningItem): bool
     {
         return $savedLearningItem->save();
     }
+
+    public static function itemExists($category, $item_id, $student_id): bool
```

Comment:
I would advise to make this non-static and instead pass an instance of the repository to the places where you need to call this method.

Reply:
I knew this was not the best way to do this, but i cannot find any alternative. In particular, I cannot find how to call this method in the view without making it static. Is there any example somewhere else in the code.

Reply:
The way you include it now, as a variable to the view is correct. Just one small error left, I think that is fixable by using the null coalescing operator `??`. I would use it like this:

```php
        return view('pages.producing.home', [
            'evaluatedTip' => $evaluatedTip ?? null,
            'itemExists' => $itemExists ?? false
        ]);
```